.github/workflows/ci.yaml

```diff
@@ -86,7 +105,7 @@ jobs:
               working-directory: ${{ matrix.folder }}
               # Don't run e2e/npm_translate_lock_auth unless there is an auth token secret; this won't
               # be set on forks
-              if: ${{ matrix.folder != 'e2e/npm_translate_lock_auth' || env.ASPECT_NPM_AUTH_TOKEN != ''}}
+              if: ${{ steps.has_test_sh.outputs.files_exists != 'true' && (matrix.folder != 'e2e/npm_translate_lock_auth' || env.ASPECT_NPM_AUTH_TOKEN != '') }}
```

Comment:
is it really useful to be so general and combine that test.sh with the matrix setup?
at least add some comment about when the test.sh outputs is expected to be true, I don't quite follow how this logic works

Reply:
Borrowed it from our bazel-examples repo; https://github.com/aspect-build/bazel-examples/blob/83d07540a160c28a6ec56c1a383dd9811022049c/.github/workflows/ci.yaml#L61

In short, if there is a test.sh file in a matrix folder then it runs that, otherwise it runs `bazel test ...` and maybe `bazel coverage ...`. Running a test.sh instead of `bazel test ...` is a useful pattern for some e2e tests; any integration test such as this one that doesn't fit nicely into a bazel test case use this pattern.

rules_ts has a slightly different variation of this check https://github.com/aspect-build/rules_ts/blob/1cbe54029c00241a0ea399b97d0b4d4175cee8a8/.github/workflows/ci.yaml#L107

Reply:
Added better comments and cleaned it up so that folders are dynamically generated as well. The `npm_translate_lock_auth` e2e is special and needs to be filtered on on fork PRs since it requires secrets.